appmap/flask.py

```diff
@@ -113,16 +117,15 @@ def before_request(self):
     def before_request_main(self, rec, request):
         Metadata.add_framework("flask", flask.__version__)
         np = None
-        # See
-        # https://github.com/pallets/werkzeug/blob/2.0.0/src/werkzeug/routing.py#L213
-        # for a description of parse_rule.
         if request.url_rule:
-            np = "".join(
-                [
-                    f"{{{p}}}" if c else p
-                    for c, _, p in parse_rule(request.url_rule.rule)
-                ]
+            # Transform requestl.url to the expected normalized-path form. For example,
+            # "/post/<username>/<post_id>/summary" becomes "/post/{{username}}/{{post_id}}/summary".
+            r = repr(request.url_rule)
+            np = NP_XLATE.sub(
+                lambda m: "{" if m["l"] else "}" if m["r"] else "",
+                NP_REGEXP.findall(r)[0],
```

Comment:
I don't think this works. I tried `r = '/post/<username>/<post_id>/summary'` — I got `IndexError: list index out of range`. I tried `r = '/post/<Rule \'username\'>/<Rule \'post_id\'>/summary'` — I got `np = 'username'`. Do we have tests for this?

Also, does this even belong in this pull request?

Reply:
Yes, there's a test for it, but I'll back and look at what's going on with.

Well, including it seemed harmless, but clearly it's not working well enough to keep. I'll take it out, and fix whatever problems it has as part of the Flask 2 work.